Data/OJS.Data.Models/Contests/Contest.cs

```diff
@@ -203,6 +203,12 @@ public bool ResultsArePubliclyVisible
         [NotMapped]
         public bool IsOnline => this.Type == ContestType.OnlinePracticalExam;
 
+        [NotMapped]
+        public bool IsOnsite => this.Type == ContestType.OnsitePracticalExam;
+
+        [NotMapped]
+        public bool IsExam => this.IsOnline || this.IsOnsite;
```

Comment:
We should restrain from easily adding properties to the entities, even more NotMapped ones. This means that the entity contains business logic which should not happen. I know this occurs in other parts of the application already but its part of the bad practices we should refactor step by step. Please use a service model or refactor the business logic.